lib/Text/PrettyPrint/Boxes.hs

```diff
@@ -78,52 +67,15 @@ module Text.PrettyPrint.Boxes
 
     ) where
 
-#if MIN_VERSION_base(4,11,0)
-import Prelude hiding ( (<>), Word )
-#elif MIN_VERSION_base(4,8,0)
-import Prelude hiding (Word)
-#else
-import Data.Foldable (Foldable (foldr))
-import Prelude hiding (foldr)
-#endif
-import Data.Foldable (toList)
-
-#if MIN_VERSION_base(4,4,0)
-import Data.String (words, unwords)
-#else
-import Data.List (words, unwords)
-#endif
-
-#ifdef OVERLOADED_STRINGS
-import Data.String (IsString(..))
-#endif
-
-import Control.Arrow ((***), first)
-import Data.List (foldl', intersperse)
-
-import Data.List.Split (chunksOf)
-
--- | The basic data type.  A box has a specified size and some sort of
---   contents.
-data Box = Box { rows    :: Int
-               , cols    :: Int
-               , content :: Content
-               }
-  deriving (Show)
+import Text.PrettyPrint.Boxes.Imports
```

Comment:
No, an `Imports` module doesn't make sense either. That's the sort of thing you want for an alternative prelude, which is really extreme overkill.

Reply:
So how do we deal with pragmas?

Reply:
What do they have to do with it?

Reply:
Pragmas are there to make possible backwards compatibility. They do it in two ways:
1. Enable definitions of newer classes such as `IsString`.
2. Manage what names are imported and from where.

Class definitions occur at one place and take one `if` pragma to disable. Imports occur in every module and may contain a larger number of `if` pragmas; they make imports a pain to look at.

Therefore we may cut the number of pragmas a developer has to look at by moving the imports into a separate module.

Reply:
I have been looking at the ways people manage this problem and I found that some use `base-compat` as a sort of _"alternate prelude"_. Do you think it makes sense to try that approach instead?